seaworthy/tests-core/test_logs.py

```diff
@@ -421,3 +434,108 @@ def test_kwargs(self):
+        """
+        logger = self.start_logging_container()
+        self.wflm(logger, EqualsMatcher('Log entry 1'), timeout=1)
+
+        early_logs = logger.logs().decode('utf8').splitlines()
+        print('early_logs:', early_logs)
+        self.assertIn('Log entry 1', early_logs)
+        self.assertNotIn('Log entry 4', early_logs)
+
+        streamed_logs = []
+        with self.assertRaises(TimeoutError):
+            for line in self.stream(logger, timeout=0.7):
+                streamed_logs.append(line.decode('utf8').rstrip())
+        print('streamed_logs:', streamed_logs)
+        self.assertNotIn('Log entry 1', streamed_logs)
+        self.assertIn('Log entry 4', streamed_logs)
+        self.assertNotIn('Log entry 9', streamed_logs)
+
+        swh_logs = []
+        for line in self.swh(logger, timeout=2):
+            swh_logs.append(line.decode('utf8').rstrip())
+        print('swh_logs:', swh_logs)
+        self.assertIn('Log entry 1', swh_logs)
+        self.assertIn('Log entry 4', swh_logs)
+        self.assertIn('Log entry 9', swh_logs)
+
+
+class TestWithFakeContainer(unittest.TestCase, FakeAndRealContainerMixin):
+    def mkcontainer(self, *args, **kw):
+        con = FakeLogsContainer(*args, **kw)
+        self.addCleanup(con.cleanup)
+        return con
+
+    def start_logging_container(self):
+        return self.mkcontainer([
+            (0.2, 'Log entry {}\n'.format(n).encode('utf8'))
+            for n in range(10)])
+
+    def wflm(self, con, matcher, timeout=0.5, **kw):
+        # Always clean up the feeder machinery when we're done. This is only
+        # necessary if we timed out, but it doesn't hurt to clean up an
+        # already-clean FakeLogsContainer.
+        try:
+            return wait_for_logs_matching(con, matcher, timeout=timeout, **kw)
+        finally:
+            con.cleanup()
+
+    def stream(self, con, timeout=1):
+        # Always clean up the feeder machinery when we're done. This is only
+        # necessary if we timed out, but it doesn't hurt to clean up an
+        # already-clean FakeLogsContainer.
+        try:
+            for line in stream_logs(con, timeout=timeout):
+                yield line
+        finally:
+            con.cleanup()
+
+    def swh(self, con, timeout=0.5, **kw):
+        # Always clean up the feeder machinery when we're done. This is only
+        # necessary if we timed out, but it doesn't hurt to clean up an
+        # already-clean FakeLogsContainer.
+        try:
+            for line in stream_with_history(con, timeout=timeout, **kw):
+                yield line
+        finally:
+            con.cleanup()
+
+
+@dockertest()
+class TestWithRealContainer(unittest.TestCase, FakeAndRealContainerMixin):
+    def setUp(self):
+        self.dh = DockerHelper()
+        self.addCleanup(self.dh.teardown)
+        self.dh.setup()
+
+    def start_logging_container(self):
+        script = '\n'.join([
+            'for n in 0 1 2 3 4 5 6 7 8 9; do',
```

Comment:
`for n in $(seq 0 9)`? (Not that it really matters)

Reply:
I don't want to rely on `seq` being available -- it's not POSIX (BSD-derived systems generally have `jot` instead) and thus may not be present in arbitrary containers.